sandstone/client/sandstone/components/filetreedirective/tests/filetree.directive.test.js

```diff
@@ -1,208 +1,421 @@
+              scope.sd = {
+                  noSelections: true,
+                  multipleSelections: false,
+                  dirSelection: false
+              };
+              var el = angular.element('<div sandstone-filetree tree-data="ctrl.treeData" leaf-level="dir" selection-desc="sd"></div>');
+              element = $compile(el)(scope);
+              scope.$digest();
+
+              // Get isolate scope
+              isolateScope = element.isolateScope();
+          });
+          it('initializeFiletree', function() {
+              // Create spies
+              spyOn(isolateScope, 'updateFiletree');
+              spyOn(isolateScope, 'deletedFile');
+              spyOn(isolateScope, 'pastedFiles');
+
+              expect(isolateScope.leafLevel).toBe("dir");
+              expect(isolateScope.selectionDesc.noSelections).toBeTruthy();
+              expect(isolateScope.selectionDesc.multipleSelections).not.toBeTruthy();
+              expect(isolateScope.selectionDesc.dirSelected).not.toBeTruthy();
+              expect(isolateScope.treeData.filetreeContents.length).toBe(2);
+          });
+          it('getDirContents', function(){
+              isolateScope.getDirContents(isolateScope.treeData.filetreeContents[0], true);
+              // First directory should have 4 files
+              expect(isolateScope.treeData.filetreeContents[0].children.length).toBe(4);
+          });
+      });
+      describe('template', function() {
+          beforeEach(function() {
+              scope.sd = {
+                  noSelections: true,
+                  multipleSelections: true,
+                  dirSelection: false
+              };
+              var el = angular.element('<div sandstone-filetree tree-data="ctrl.treeData" leaf-level="dir" selection-desc="sd"></div>');
+              element = $compile(el)(scope);
+              scope.$digest();
+
+              // Get isolate scope
+              isolateScope = element.isolateScope();
+          });
+          var getExpandedNodes = function(tpl) {
+              var matches = tpl.match(/fa-caret-down/g);
+              if(!matches) {
+                  return [];
+              }
+              return matches;
+          };
+
+          var getContractedNodes = function(tpl) {
+              var matches = tpl.match(/fa-caret-right/g);
+              if(!matches) {
+                  return [];
+              }
+              return matches;
+          };
+
+          var getSelectedNodes = function(tpl) {
+              var matches = tpl.match(/tree-selected/g);
+              if(!matches) {
+                  return [];
+              }
+              return matches;
+          };
+
+          var getTemplate = function() {
+              scope.$digest();
+              return element.html();
+          };
+
+          describe('multipleSelections: false', function() {
+              it('should show correctly expanded nodes and leaf nodes', function() {
```

Comment:
If the only thing that distinguishes this test from the one below bearing the same name is multiselection, they should be grouped together into a single integration test with multiselection toggled in that test. This makes the expectations themselves more readable, and makes it easier to browse behavior covered by tests.